checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,334 @@
+package org.checkerframework.checker.calledmethods;
+
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.Tree;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.builder.qual.ReturnsReceiver;
+import org.checkerframework.checker.calledmethods.framework.AutoValueSupport;
+import org.checkerframework.checker.calledmethods.framework.FrameworkSupport;
+import org.checkerframework.checker.calledmethods.framework.FrameworkSupportUtils;
+import org.checkerframework.checker.calledmethods.framework.LombokSupport;
+import org.checkerframework.checker.calledmethods.qual.CalledMethods;
+import org.checkerframework.checker.calledmethods.qual.CalledMethodsBottom;
+import org.checkerframework.checker.calledmethods.qual.CalledMethodsPredicate;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.common.accumulation.AccumulationAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.value.ValueAnnotatedTypeFactory;
+import org.checkerframework.common.value.ValueChecker;
+import org.checkerframework.common.value.ValueCheckerUtils;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.javacutil.ElementUtils;
+import org.checkerframework.javacutil.TreeUtils;
+
+/** The annotated type factory for the Called Methods checker. */
+public class CalledMethodsAnnotatedTypeFactory extends AccumulationAnnotatedTypeFactory {
+
+    /** {@link java.util.Collections#singletonList} is treated specially by the EC2 logic. */
+    private final ExecutableElement collectionsSingletonList;
```

Comment:
Is it a meaningful performance optimization to keep this in a field?

Reply:
I don't have any evidence that it is. It's in a field now for code readability, I think.

Reply:
Ok.  Since we don't create a lot of type factory objects this is fine